src/main/java/org/apache/mesos/scheduler/recovery/SimpleRecoveryPlanManager.java

```diff
@@ -0,0 +1,325 @@
+
+        final OfferRequirementProvider offerRequirementProvider = new DefaultOfferRequirementProvider();
+
+        // If recovery candidates are not part of a different PlanManager's plan, generate a recovery block for them
+        // and mark it as pending. Tasks that are still in terminal state, add them to plan directly
+        for (Map.Entry<String, Protos.TaskInfo> candidate : recoveryCandidates.entrySet()) {
+            final String name = candidate.getKey();
+            final Protos.TaskInfo taskInfo = candidate.getValue();
+            try {
+                final DefaultTaskSpecification taskSpecification = DefaultTaskSpecification.create(taskInfo);
+                final List<RecoveryRequirement> recoveryRequirements;
+                if (FailureUtils.isLabeledAsFailed(taskInfo) || failureMonitor.hasFailed(taskInfo)) {
+                    recoveryRequirements = offerReqProvider.getPermanentRecoveryRequirements(Arrays.asList(taskInfo));
+                } else {
+                    recoveryRequirements = offerReqProvider.getTransientRecoveryRequirements(Arrays.asList(taskInfo));
+                }
+
+                newRecoveryPhaseBuilder.addBlock(new DefaultRecoveryBlock(
+                        taskSpecification.getName(),
+                        offerRequirementProvider.getExistingOfferRequirement(taskInfo, taskSpecification),
+                        Status.PENDING,
+                        recoveryRequirements.get(0),
+                        launchConstrainer));
+            } catch (InvalidTaskSpecificationException | InvalidRequirementException e) {
+                LOGGER.error("Error adding recovery plan for task: " + name, e);
+            }
+        }
+
+        // For blocks in plan, that are not part of recovery candidates, determine their status,
+        // using the taskInfo status
+        if (plan != null) {
+            final List<? extends Phase> phases = plan.getPhases();
+            if (CollectionUtils.isNotEmpty(phases)) {
+                // Simple plan only deals with a single recovery phase. This makes it explicit.
+                final Phase phase = phases.get(0);
+                for (Block block : phase.getBlocks()) {
+                    final String taskName = block.getName();
+                    // Ignore blocks already added to the new plan.
+                    if (!recoveryCandidates.containsKey(taskName)) {
+                        final Optional<Protos.TaskStatus> taskStatus = stateStore.fetchStatus(taskName);
+                        if (taskStatus.isPresent()) {
+                            block.update(taskStatus.get());
+                            newRecoveryPhaseBuilder.addBlock(block);
+                        }
+                    }
+                }
+            }
+        }
+
+        final DefaultPhase recoveryPhase = newRecoveryPhaseBuilder.build();
+        plan = DefaultPlan.fromArgs(recoveryPhase);
+        phaseStrategy = new RandomRecoveryStrategy(recoveryPhase);
```

Comment:
Could break out the recovery strategy into a `protected` member function. If someone wants to use a custom strategy they can then extend the class and override that function